### PR TITLE
Add docker and travis scripts for running tests

### PR DESCRIPTION
**Pull request by harshitbansal05**

### CODE REVIEW

**Comment by zhangcandrew** [comment]
Thanks for this @harshitbansal05! 

Are you able to provide any screenshots of this working locally or a description of how this is working/what the moving parts to it?

File: test/test_modified_scripts.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        print("Engine, Dataset, Error")
+        for error in errors:
+            print(error)
+        exit(1)
+    else:
+        print("All tests passed. All scripts are updated to latest version.")
+        exit(0)
+
+if __name__ == "__main__":
+    install_modified()
```
Comment (harshitbansal05) [inline test/test_modified_scripts.py]: This is the same as the `test_modified_scripts.py` present in `retriever`, where it attempts to install the modified dataset scripts in various engines.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      - ./cli_tools/postgis-data:/var/lib/postgresql
+      - ./cli_tools/:/backups
+    container_name: pgdb_recipes_container
+    environment:
+      - POSTGRES_DB=testdb_retriever
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASS=Password12!
+      - POSTGRES_MULTIPLE_EXTENSIONS=postgis,postgis_topology
+      - ALLOW_IP_RANGE=0.0.0.0/0
+    restart: on-failure
```
Comment (harshitbansal05) [inline docker-compose.yml:16]: The container for the postgres environment, with all the required extensions.

**Comment by zhangcandrew** [comment]
Thanks, have you been able to test this out yourself locally?

Something like this may be able to help you: https://stackoverflow.com/questions/21053657/how-to-run-travis-ci-locally

**Comment by harshitbansal05** [comment]
@zhangcandrew I tested it locally and it worked fine for me.

**Comment by zhangcandrew** [comment]
Cool, do you think you can add screenshots?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+language: python
+sudo: required
+services:
+  - docker
+
+script:
+  - docker-compose run --service-ports python_recipes pytest -v -k modified
```
Comment (henrykironde) [inline .travis.yml]: Remove the  this part `-k modified`.

**Review by henrykironde — COMMENTED** [review]
Look good to me after the suggested changes

File: .travis.yml (hunk 3 above)
Comment (harshitbansal05) [reply .travis.yml]: okay, would do so.

**Comment by henrykironde** [comment]
Any updates on this @harshitbansal05 ?

**Comment by harshitbansal05** [comment]
@henrykironde done.

**Comment by henrykironde** [comment]
@harshitbansal05 looks good. Don't you think we should be installing retriever from master?
`RUN pip3 install git+....retriever...`

**Comment by harshitbansal05** [comment]
We can do, but I don't think it would really matter. After all, we only need to install datasets using retriever.

**Comment by henrykironde** [comment]
Thanks @harshitbansal05.

**Pull request merged**